### PR TITLE
Created Dropdown Options

### LINKED ISSUE #867: [Feature request] navbar dropdown
**Is your feature request related to a problem? Please describe.**

I want to add more entries to the nav bar but don't want to create too much visual cluster. A drop down would be nice, as it allows me to group multiple navbar links together, taking less space by default.

**Describe the solution you'd like**

```ts
const headerNavLinks = [
  { href: '/', title: 'Home' },
  { href: '/blog', title: 'Blog' },
  { href: '/tags', title: 'Tags' },
  { title: 'Dropdown 1', children: {
href: '/link1', title: 'title 1',
href: '/link2', title: 'title 2',
} // this becomes a dropdown navbar item
]
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by pratikkabade**
## `Enhancement`

Fixes #867 

Implemented dropdown options in the navbar to allow users to hide additional options

### Visuals

Screenshot 2024-04-17 23 33 45

### CODE REVIEW

**Comment by pratikkabade** [comment]
@timlrx any problem with pr?
As you haven't yet merged it

**Comment by timlrx** [comment]
@pratikkabade the build failed. Could you run it through prettier and push the updated changes?

**Comment by pratikkabade** [comment]
I thought it failed because of permission youve set, sure I'll give it a try
Do I have to create another pr?

**Comment by timlrx** [comment]
> Inspect

It's mostly prettier related. You should have initialized the husky git pre-commit hooks which will automatically run it through eslint and prettier, but since the code is already pushed, you can just run `prettier --write` and push the formatted codes.

CleanShot 2024-04-22 at 11 44 25@2x

**Comment by pratikkabade** [comment]
@timlrx I've made `prettier formatting`, please review it

**Comment by pratikkabade** [comment]
@timlrx any problem with PR?
why haven't you merged it yet?

**Comment by timlrx** [comment]
@pratikkabade, the desktop version in good. The mobile version needs to be updated. I was thinking either to list it out but indent it like Dribble or making it an expandable dropdown like Figma. Any preference?
CleanShot 2024-04-30 at 22 29 26@2x
CleanShot 2024-04-30 at 22 31 09@2x

File: data/headerNavLinks.ts [hunk 1]
```diff
@@ -6,4 +6,12 @@ const headerNavLinks = [
   { href: '/about', title: 'About' },
 ]
 
+export const headerNavOptions = {
```
Comment (timlrx) [inline data/headerNavLinks.ts:9]: No need for an additional field. You can modify the original headerNavLink to something like this:
```
const headerNavLinks = [
  { href: '/', title: 'Home' },
  { href: '/blog', title: 'Blog' },
  { href: '/tags', title: 'Tags' },
  { title: 'Others': children: [
    { href: '/projects', title: 'Projects' },
    { href: '/about', title: 'About' },
  ]
]
```

File: components/Header.tsx [hunk 2]
```diff
@@ -37,6 +38,7 @@ const Header = () => {
               {link.title}
             </Link>
           ))}
+        <NavOptions />
```
Comment (timlrx) [inline components/Header.tsx:41]: You can move the rendering of all the `headerNavLinks` to `NavOptions`

File: data/headerNavLinks.ts (hunk 1 above)
Comment (pratikkabade) [reply data/headerNavLinks.ts:9]: Got it I'll see if i could do that and will change

**Pull request closed without merging**